app/models/portal/learner.rb

```diff
@@ -1,5 +1,9 @@
 class Portal::Learner < ActiveRecord::Base
+  include Rails.application.routes.url_helpers
+
   self.table_name = :portal_learners
+
+  KEY_PREFIX = 'key:'
```

Comment:
I realize that using `:` is technically OK in the path of a URL. But it does have a restriction because it can't be used in the first part of the path otherwise it is interpreted as being part of the protocol. That shouldn't happen in this case, but it still seems better to avoid using `:` to keep things simple.
I'd suggest switching the prefix to `key-` instead of `key:`. Would that cause a problem?